File: src/Practice/CozyCrawler.Runner/BlockedUrl2UrlRunner.cs
using CozyCrawler.Interface;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CozyCrawler.Runner
{
    public class BlockedUrl2UrlRunner : IUrl2UrlRunner
    {
        IUrlIn to_;
        IUrl2Url p_;

        public void SetProcessor(IUrl2Url p)
        {
            p_ = p;
            if (to_ != null)
            {
                p_.To(to_);
            }
        }

        public void OnNewUrl(string url)
        {
            p_?.OnNewUrl(url);
        }

        public void Start()
        {
        }

        public void Stop()
        {
        }

        public void To(IUrlIn to)
        {
            to_ = to;
            p_?.To(to_);
        }
    }
}

File: src/Practice/CozySeeker/Runner/BlockedUrl2UrlRunner.cpp
#include "BlockedUrl2UrlRunner.h"

NS_BEGIN

BlockedUrl2UrlRunner::BlockedUrl2UrlRunner()
{

}

BlockedUrl2UrlRunner::~BlockedUrl2UrlRunner()
{

}

void BlockedUrl2UrlRunner::SetProcessor(IUrl2UrlPtr p)
{
    m_Process = p;
    if (m_urlIn != nullptr)
    {
        p->To(m_urlIn);
    }
}

void BlockedUrl2UrlRunner::OnNewUrl(StrPtr url)
{
    if (m_Process != nullptr)
    {
        m_Process->OnNewUrl(url);
    }
}

void BlockedUrl2UrlRunner::Start()
{
    if (m_Process != nullptr)
    {
        m_Process->Start();
    }
}

void BlockedUrl2UrlRunner::Stop()
{
    if (m_Process != nullptr)
    {
        m_Process->Stop();
    }
}

void BlockedUrl2UrlRunner::To(IUrlInPtr to)
{
    m_urlIn = to;
    if (m_Process != nullptr)
    {
        m_Process->To(m_urlIn);
    }
}

NS_END